src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,622 @@
+	*     times: [ { weekDay, startTime, endTime, room } ],
+	*     state: 'new',
+	*   }
+	*/
+	async fetchMetaData() {
+		const intermediateData = {};
+		
+		intermediateData.currentSchoolYear = await this.getCurrentSchoolyear();
+
+		// To iterate over either concept
+		intermediateData.classes = await this.getClasses(intermediateData.currentSchoolYear.id);
+		intermediateData.teachers = await this.getTeachers();
+		intermediateData.rooms = await this.getRooms();
+		// intermediateData.subjects = await this.getSubjects(); // currently not required
+
+		// intermediateData.timeGrid = await this.getTimegrid(); // currently not required
+
+		intermediateData.courses = [];
+
+		// Iteration approaches currently implemented: teachers
+		if (intermediateData.teachers !== undefined) { // Iterate over teachers
+			for (const teacher of intermediateData.teachers) {
+				const currentTeacherName = `${teacher.foreName} ${teacher.longName}`;
+
+				let timetable = await this.getCustomizableTimeTableFor(2, teacher.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					roomFields: ['id', 'longname'],
+					teacherFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				let filteredTimetable = timetable.filter(entry => !(entry.te.length === 1
+					&& entry.te[0].id === teacher.id
+					&& entry.kl.length > 0
+					&& entry.ro.length === 1
+					&& entry.su.length === 1));
+				if (filteredTimetable.length > 0) {
+					filteredTimetable = filteredTimetable.map(entry => {
+						return {
+							teachers: entry.te.map(teacher => teacher.longname),
+							teacherValid: entry.te[0].id === teacher.id,
+							rooms: entry.ro.map(room => room.longname),
+							roomValid: entry.ro.length === 1,
+							classes: entry.kl.map(klass => klass.longname),
+							classesValid: entry.kl.length > 0,
+							subjects: entry.su.map(subject => subject.longname),
+							subjectValid: entry.su.length === 1
+						};
+					});
+					this.logger.warning(`Ignored timetable entries from WebUntis import`, { ignored: filteredTimetable });
+				}
+
+				timetable = timetable.filter(entry => entry.te.length === 1
+					&& entry.te[0].id === teacher.id
+					&& entry.kl.length > 0
+					&& entry.ro.length === 1
+					&& entry.su.length === 1);
+				/* END TODO: Check for change */
+
+				const compareArraysOfString = (a1, a2) => {
+					if (a1.length !== a2.length) {
+						return false;
+					}
+
+					for (let i = 0; i < a1.length; ++i) {
+						if (a1[i] !== a2[i]) {
+							return false;
+						}
+					}
+
+					return true;
+				};
+	
+				for (const entry of timetable) {
+					const classNames = entry.kl.map(k => k.longname);
+
+					// Select course with matching teacher, subject, and participating classes
+					let course = intermediateData.courses.find(course => {
```

Comment:
Codacy found an issue: [Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-server/pullRequest?prid=6045816)